consumer/src/test/kotlin/integrations/KafkaEventSinkIT.kt

```diff
@@ -0,0 +1,103 @@
+import org.neo4j.test.TestGraphDatabaseFactory
+import org.testcontainers.containers.KafkaContainer
+import streams.StreamsEventSinkQueryExecutionTest
+import java.util.*
+import java.util.concurrent.ThreadLocalRandom
+import kotlin.test.assertEquals
+
+class KafkaEventSinkIT {
+    companion object {
+        /**
+         * Kafka TestContainers uses Confluent OSS images.
+         * We need to keep in mind which is the right Confluent Platform version for the Kafka version this project uses
+         *
+         * Confluent Platform | Apache Kafka
+         *                    |
+         * 4.0.x	          | 1.0.x
+         * 4.1.x	          | 1.1.x
+         * 5.0.x	          | 2.0.x
+         *
+         * Please see also https://docs.confluent.io/current/installation/versions-interoperability.html#cp-and-apache-kafka-compatibility
+         */
+        private const val confluentPlatformVersion = "4.0.2"
+        @ClassRule
+        @JvmField
+        val kafka = KafkaContainer(confluentPlatformVersion)
+    }
+
+    private lateinit var db: GraphDatabaseService
+
+    private val objectMapper: ObjectMapper = ObjectMapper()
+
+    @Before
+    fun setUp() {
+        db = TestGraphDatabaseFactory()
+                .newImpermanentDatabaseBuilder()
+                .setConfig("kafka.bootstrap.servers", kafka.bootstrapServers)
+                .setConfig("streams.sink.topic.shouldWriteCypherQuery", "MERGE (n:Label {id: event.id})\n" +
+                        "\tON CREATE SET n += event.properties")
+                .newGraphDatabase()
+    }
+
+    @After
+    fun tearDown() {
+        db.shutdown()
+    }
+
+    @Test
+    fun shouldWriteDataFromSink() = runBlocking {
+        val props = Properties()
+        props[ProducerConfig.BOOTSTRAP_SERVERS_CONFIG] = kafka.bootstrapServers
+        props["group.id"] = "neo4j"
+        props[ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG] = LongSerializer::class.java
+        props[ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG] = ByteArraySerializer::class.java
+        val topics = listOf("shouldWriteCypherQuery")
+        AdminClient.create(props).use { client -> client.createTopics(
+                topics.map { topic -> NewTopic(topic, 1, 1.toShort()) }) }
+        val producer = KafkaProducer<Long, ByteArray>(props)
+        val dataProperties = mapOf("prop1" to "foo", "bar" to 1)
+        val data = mapOf("id" to 1, "properties" to dataProperties)
+        val job = GlobalScope.launch {
+            var partition = ThreadLocalRandom.current().nextInt(1)
+            var producerRecord = ProducerRecord(topics[0], partition, System.currentTimeMillis(), 1L,
+                    objectMapper.writeValueAsBytes(data))
+            producer.send(producerRecord)
+        }
+        job.join()
+        delay(3000)
+        db.beginTx().use {
+            db.execute("MATCH (n:Label) RETURN n")
+                    .columnAs<Node>("n").use {
+                        assertEquals(true, it.hasNext())
```

Comment:
we can just run the exact query we're looking for and check the count

MATCH (n:Label) WHERE properties(n) = $props RETURN count(*)